subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidatePlugins.java

```diff
@@ -100,10 +97,18 @@ private static CharSequence toMessageList(List<String> problemMessages) {
         return builder;
     }
 
-    private static List<InvalidUserDataException> toExceptionList(List<String> problemMessages) {
-        return problemMessages.stream()
-            .map(InvalidUserDataException::new)
-            .collect(Collectors.toList());
+    private String buildValidationErrorMessage(List<String> problemMessages) {
```

Comment:
Should building the message go to the exception maybe? So you'd have a constructor taking a list of messages?